WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -120,16 +122,62 @@ extension LocationDetailViewController: UITableViewDelegate, UITableViewDataSour
     }
 }
 
-//extension LocationDetailViewController: UICollectionViewDataSource, UICollectionViewDelegate {
-//
-//    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-//        return weatherDetail.dailyWeatherData.count
-//    }
-//
-//    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-//        let hourlyCell = collectionView.dequeueReusableCell(withReuseIdentifier: "HourlyCell", for: indexPath)
-//        return hourlyCell
-//    }
-//
-//
-//}
+extension LocationDetailViewController: CLLocationManagerDelegate {
+    
+    func getLocation() {
+        locationManager = CLLocationManager()
+        locationManager.delegate = self
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus){
+        print("checking authentication status")
+        handleAuthorizationStatus(status: status)
+    }
+    
+    func handleAuthorizationStatus(status: CLAuthorizationStatus) {
+        switch status {
+            
+        case .notDetermined:
+            locationManager.requestWhenInUseAuthorization()
+        case .restricted:
+            self.oneButtonAlert(title: "location service denied", message: "parental control resricting location use")
+        case .denied:
+            break
+        case .authorizedAlways, .authorizedWhenInUse:
+            locationManager.requestLocation()
+        
+        @unknown default:
+            print("unknown case of status in handleAthenticalStatus\(status)")
+        }
+        
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        let currentLocation = locations.last ?? CLLocation()
+        print (" 😇😇😇😇 current location is \(currentLocation.coordinate.latitude), \(currentLocation.coordinate.longitude)")
+        let geocoder = CLGeocoder()
+        geocoder.reverseGeocodeLocation(currentLocation){ ( placemarks, error) in
+            var locationName = ""
+            if placemarks != nil {
```

Comment:
When dealing with optionaly we use optional unwrapping things like `!= nil` should never exist in your swift code rather its should be an `if let placemarks {  }` or `guard let placemarks else { return }`